routes/users.go

```diff
@@ -152,6 +153,51 @@ func PostLoginHandler(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+func PutFavoriteHandler(w http.ResponseWriter, r *http.Request) {
+
+	// Checks for food truck ID
+	params := mux.Vars(r)
+	foodTruckID, foodTruckIDExists := params["foodTruckID"]
+	if !foodTruckIDExists {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Get user from context
+	userID, userLoggedIn := r.Context().Value(middleware.UserKey).(string)
+
+	// Check for a user, or if the user agent is from the scraper
+	if !userLoggedIn {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	var updateOperator string
+
+	// Determine if an add or delete
+	action := r.URL.Query().Get("action")
+	if action == "add" {
```

Comment:
Mmm, I'd say a different route for add and deleting favorites is favorable here, query parameters are really just for GET routes

Reply:
hmmmm well it works and andrea says that route is good

Reply:
so i think ill keep it